scripts/network.js

```diff
@@ -288,6 +290,91 @@ export async function getStakingRewards() {
     }
 }
 
+export async function getMasternodeRewards() {
+    if (!networkEnabled || masterKey == undefined) return;
+    doms.domGuiMasternodeLoadMoreIcon.style.opacity = 0.5;
+    const stopAnim = () => (doms.domGuiMasternodeLoadMoreIcon.style.opacity = 1);
+    const nHeight = masternodeRewards.length
+        ? masternodeRewards[masternodeRewards.length - 1].blockHeight
+        : 0;
+    let mapPaths = new Map();
+    const txSum = (v) =>
+        v.reduce(
+            (t, s) =>
+                t +
+                (s.addresses
+                    .map((strAddr) => mapPaths.get(strAddr))
+                    .filter((v) => v).length && s.addresses.length === 2
+                    ? parseInt(s.value)
+                    : 0),
+            0
+        );
+    let cData;
+    if (masterKey.isHD) {
```

Comment:
There's no need to check all the addresses, we already know the masternode address

Reply:
Pushed to use local storage, but seems off from what I thought of to do